### PR TITLE
Fix App Store issue while uploading the app to App Store because of App Icons are invalid format

### PR DESCRIPTION
**Pull request by devadiab**
Fix App Store issue while uploading the app to App Store because of App Icon has invalid format and they have to be stripped from alpha channel.

The previous image was generated using default settings of .NET graphics. I have fixed it and adding additional app icon sizes by simply setting `CompositingMode` attribute for .NET graphics to `CompositingMode.SourceOver`

for further information see this example of XCode error message after archiving the App

Xcode error screen

Also it's listed here in Apple's guidelines for [Human Interface Guidelines ](https://developer.apple.com/ios/human-interface-guidelines/icons-and-images/app-icon/):

> Keep the background simple and avoid transparency. Make sure your icon is opaque, and don’t clutter the background. Give it a simple background so it doesn’t overpower other app icons nearby. You don’t need to fill the entire icon with content.

### CODE REVIEW

**Comment by devadiab** [comment]
I will also add new App Icon sizes in Change log file 👍

File: src/engine/Uno.ProjectFormat/PropertyDefinitions.cs [hunk 1]
```diff
@@ -53,24 +53,31 @@ public class PropertyDefinitions : Dictionary<string, Tuple<PropertyType, string
...
+            {"iOS.Icons.iPad_40_1x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_40_2x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_50_1x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_50_2x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_72_1x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_72_2x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_76_1x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_76_2x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iPad_83.5_2x",PropertyType.Path, "$(Icon)"},
+            {"iOS.Icons.iOS-Marketing_1024_1x",PropertyType.Path,"$(Icon)"},
```
[the remark this answers is not in the document]
Comment (devadiab) [reply src/engine/Uno.ProjectFormat/PropertyDefinitions.cs]: Sure, I will put back the space again , I have added at least 7 lines on it

File: src/compiler/Uno.Compiler.Core/Syntax/UxlProcessor.cs [hunk 2]
```diff
@@ -628,6 +629,8 @@ void WriteImageFile(ImageFile f)
                         {
                             graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
                             graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
+                            graphics.Clear(Color.White);
+                            graphics.CompositingMode = CompositingMode.SourceOver;
```
[the remark this answers is not in the document]
Comment (devadiab) [reply src/compiler/Uno.Compiler.Core/Syntax/UxlProcessor.cs]: Yes it does, I was trying to draw rectangle behind the image and many other ways but all failed and only this works !

File: src/compiler/Uno.Compiler.Core/Syntax/UxlProcessor.cs [hunk 3]
```diff
@@ -628,6 +629,8 @@ void WriteImageFile(ImageFile f)
                         {
                             graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
                             graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
+                            graphics.Clear(Color.White);
```
[the remark this answers is not in the document]
Comment (devadiab) [reply src/compiler/Uno.Compiler.Core/Syntax/UxlProcessor.cs]: Really, I didn't figure it out, I will add the condition for the platform which is "iOS".

**Comment by devadiab** [comment]
> This makes sure that if the input-image has an alpha channel, the output image will also. This leaves the "should my icon have an alpha-channel"-question up to the user, rather than a hard decision by us...

Apple's policy obligate us to do that, because it's not accepting the app to be uploaded to iTunesConnect even for TestFlight if the icon image has alpha channel !

**Comment by kusma** [comment]
>> This makes sure that if the input-image has an alpha channel, the output image will also. This leaves the "should my icon have an alpha-channel"-question up to the user, rather than a hard decision by us...

> Apple's policy obligate us to do that, because it's not accepting the app to be uploaded to iTunesConnect even for TestFlight if the icon image has alpha channel !

I don't think that's quite right. Our default icon does not have an alpha-channel AFAIK, and if the user has specified a custom icon with an alpha-channel, this is a user-error IMO, which can easily be fixed in the user-project. So I don't think we need to handle the case where the image has an alpha-channel other than forwarding that alpha channel.

Removing the alpha channel, by blending the image on top of a background color makes assumptions about what the desired background color is, which is rather arbitrary. I'd rather not make arbitrary decisions like that.

**Pull request closed without merging**